[next] Comment Editing

wyattjoh opened the pull request:
## What does this PR do?

Adds the `editComment` mutation and `editing` metadata.

## How do I test this PR?

1. `createComment` mutation
2. `editComment` mutation
3. `comment` query to view `editing` metadata

wyattjoh committed:
feat: initial impl of comment edits

wyattjoh committed:
feat: added edit metadata

wyattjoh committed:
Merge branch 'next' into next-comment-edit

alinhle commented on src/core/server/graph/tenant/mutators/comment.ts:
```diff
@@ -1,11 +1,14 @@
 export default (ctx: TenantContext) => ({
-  create: (input: GQLCreateCommentInput): Promise<Comment> => {
-    return create(
+  create: (input: GQLCreateCommentInput): Promise<Comment | null> =>
```
Why would it return `null`?

alinhle commented on src/core/server/graph/tenant/mutators/comment.ts:
```diff
@@ -16,6 +19,17 @@ export default (ctx: TenantContext) => ({
-    );
-  },
+    ),
+  edit: (input: GQLEditCommentInput): Promise<Comment | null> =>
```
Same question, why would it return null?

alinhle commented on src/core/server/graph/tenant/schema/schema.graphql:
```diff
@@ -470,13 +482,18 @@ type Comment {
+  """
+  editing returns details about the edit status of a Comment.
+  """
+  editing: EditInfo
```
What do you think of adding it directly to `Comment` instead of using a separate type `EditInfo`? Makes it slightly easier to consume on the client. Just a thought.

alinhle replied:
Besides that, shouldn't this be `EditInfo!`?

wyattjoh replied:
I can change it to `EditInfo!`, but without repeating the same logic twice, shouldn't it be the same easiness level to access `comment.editInfo.?` than accessing `comment.?`?

alinhle replied:
I guess it's fine. Optimistic updates are a bit more code first getting the comment, then getting the linked record and then setting the value. The more nested it is, the more cumbersome it is with the vanilla relay api, one level should be ok.

wyattjoh replied:
Sounds good! Just don't take a look at the Karma API :P

wyattjoh committed:
Merge branch 'next' into next-comment-edit

wyattjoh committed:
Merge branch 'next' into next-comment-edit

wyattjoh committed:
review: switched out explcit null + nullable types

kgardnr committed:
Merge branch 'next' into next-comment-edit

wyattjoh committed:
Merge branch 'next' into next-comment-edit

wyattjoh committed:
fix: changed commentEdge -> edge, test comments

wyattjoh committed:
Merge branch 'next' into next-comment-edit

wyattjoh committed:
[next] Tasks (#1777)

* feat: initial support for synced tenants

* fix: cleanup

* fix: logger now respects logging level

* fix: cache now ignores updates issued from itself

* feat: print subscriber count

* feat: initial moderation + validation for new comments

* fix: added Promiseable type

* feat: initial actions impl

* feat: more moderation phases

* fix: handle settings inheritence

* fix: moved settings into new file

* fix: defaults and documentation

* fix: replace merge with object spread

* feat: added integration with akismet

* fix: support tenant cache for oidc strategy

* fix: fixed compile

* fix: import ordering

* feat: added bull for queue support

* feat: support for scraping

* fix: fixes for scraper

- Implemented simple metascraper replacement (to resolve security advisory
  warning)
- Implemented simle dotize replacement (to resolve not
  working version that couldn't handle date objects)
- Plugged in asset scraping to asset creation process

* fix: handles array values

* feat: added initial scraper implementation

* feat: seperate queues but share config

* fix: simplified auth data access

* feat: moved more settings into the graph

* feat: improved mailer design

* fix: fixed issue with dotize

* fix: fixed some issues with adapter

* fix: queue cleanup

* feat: added organizationName to Tenant

* feat: email rendering

* review: support es6 imports

* fix: restore old ci step

* fix: adjusted logging messages

wyattjoh committed:
fix: linting

wyattjoh committed:
Merge branch 'next' into next-comment-edit
